API/Controllers/FileController.cs

```diff
@@ -0,0 +1,213 @@
+* along with this program, in the LICENSE.md file in the root project directory.
+* If not, see https://www.gnu.org/licenses/lgpl-3.0.txt
+*/
+
+using API.Extensions;
+using API.Resources;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Models;
+using Models.Defaults;
+using Services.Services;
+using System;
+using System.IO;
+using File = Models.File;
+
+namespace API.Controllers
+{
+    /// <summary>
+    ///     File controller for files
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class FileController : ControllerBase
+    {
+
+        private readonly IFileService fileService;
+        private readonly IMapper mapper;
+        private readonly IProjectService projectService;
+        private readonly IUserService userService;
+        private readonly IFileUploader fileUploader;
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileController"/> class.
+        /// </summary>
+        /// <param name="fileService">The file service.</param>
+        /// <param name="mapper">The mapper.</param>
+        /// <param name="projectService">The Project service</param>
+        /// <param name="userService">The User service</param>
+        /// /// <param name="fileUploader">The file uploader extension</param>
+        public FileController(IFileService fileService, IMapper mapper, IProjectService projectService, IFileUploader fileUploader, IUserService userService)
+        {
+            this.fileService = fileService;
+            this.mapper = mapper;
+            this.projectService = projectService;
+            this.userService = userService;
+            this.fileUploader = fileUploader;
+        }
+
+        /// <summary>
+        /// Get all files
+        /// </summary>
+        /// <returns>A response and list of files.</returns>
+        [HttpGet]
+        [Authorize]
+        public async Task<IActionResult> GetFilesAsync()
+        {
+            IEnumerable<File> files = await fileService.GetFilesAsync();
+            if(!files.Any())
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting files.",
+                    Detail = "The database does not contain any files.",
+                    Instance = "47525791-57C4-4DE2-91B1-90086D893112"
+                };
+                return NotFound(problem);
+            }
+
+            return Ok(mapper.Map<IEnumerable<File>, IEnumerable<FileResourceResult>>(files));
+        }
+
+        /// <summary>
+        /// Uploads a single file
+        /// </summary>
+        /// <returns>HTTP Response</returns>
+        [HttpPost]
+        [Authorize]
+        public async Task<IActionResult> UploadSingleFile([FromForm] FileResource fileResource)
```

Comment:
This also requires to send all kinds of user data, these are required within the nested User object. If you go to swagger you see that those fields are required, while that info is gotten from the getcontextuser method on HttpContext.

Reply:
Maybe this happens because it asks for multipart-form-data? 

On delete is already check if the user owns the file or is administrator: https://gyazo.com/ecb4118f25c037fa58440b47d9f0baae
Although I see it should be " || " instead of "&& " right?